File: infrastructure/src/main/java/com/study/platform/global/oauth/kakao/dto/KakaoIdTokenPayload.java
package com.study.platform.global.oauth.kakao.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonIgnoreProperties(ignoreUnknown = true)
public record KakaoIdTokenPayload(

        String sub,

        String email,

        @JsonProperty("nickname")
        String nickname
) {}

File: infrastructure/src/main/kotlin/com/study/platform/global/oauth/kakao/dto/KakaoIdTokenPayload.kt
package com.study.platform.global.oauth.kakao.dto

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
data class KakaoIdTokenPayload @JsonCreator constructor(
    @JsonProperty("sub")
    val sub: String,
    @JsonProperty("email")
    val email: String,
    @JsonProperty("nickname")
    val nickname: String
)
